### DISCUSSION TITLE
Cannot DNS resolve error after upgrade to 2.6

### DISCUSSION THREAD

**Discussion by timkolloch** [qdm12/ddns-updater#679] [Q&A] [answered]
I really do not know what happend and I do not think that anything else in my network changed. Setting the log level to debug did not bring any more information so I do not really know what to do.

I am using strato as my provider. A few days ago I accidentaly (yes I was stupid) requested an owner change which I immediately declined but maybe this has some kind of "retention period"?

My DNS is a pihole running on 192.168.178.8:53

Here is the error message: 

`2024-03-25T16:19:35Z WARN cannot DNS resolve <mydomain> after 5 tries: lookup <mydomain> on 192.168.178.8:53: read udp 172.17.0.2:40898->192.168.178.8:53: i/o timeout`

Thanks in advance for every help!

EDIT:

That's the error on the web ui:
`request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`

**Comment by qdm12** [comment, marked as answer]
`WARN cannot DNS resolve <mydomain> after 5 tries: lookup <mydomain> on 192.168.178.8:53: read udp 172.17.0.2:40898->192.168.178.8:53: i/o timeout` in particular `i/o timeout` means your pihole is not responding 🤔 Perhaps try it by setting `RESOLVER_ADDRESS=1.1.1.1` to check it's not pihole at fault?

**Reply by timkolloch** [reply]
Seems to have been a pihole problem. I really thought nothing has changed there, will look into that. Thank you very much for the hint!

**Comment by timkolloch** [comment]
I am sorry but I have to reopen this. 

I have now stated the container with `RESOLVER_ADDDRESS=1.1.1.1:53` and it seemed to work for a day or two. 
However now I get this error:

`2024-03-30T12:46:59Z ERROR doing http request: Get "mydomain:***@dyndns.strato.com/nic/update?hostname=myhostname": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`

If I shut down my pihole container I get 

`2024-03-30T12:50:16Z ERROR doing http request: Get "mydomain:***@dyndns.strato.com/nic/update?hostname=myhostname": dial tcp: lookup dyndns.strato.com on 192.168.178.8:53: read udp 172.17.0.2:44409->192.168.178.8:53: read: connection refused`

Which for me indicates that the container is still trying to reach my local DNS instead of using `1.1.1.1` directly. Is there something missing in my configuration or do I have to change something?

If I look into the pihole query list I do see the requests to "dyndns.strato.com" and they are not blocked, so general connection between my DDNS-updater and my pihole seems to work as well.

It is completely fine for me if the DDNS updater does not use my pihole and instead uses `1.1.1.1` directly but I do not get why this it not working.

Any help is appreciated, thanks in advance!

**Reply by qdm12** [reply]
`RESOLVER_ADDRESS` is actually only used to resolve your domain names, not for anything else. Readme improved for this in 99d670f7f992e3db69edecd403b3f209053fb5dd.

So in your case `mydomain:***@dyndns.strato.com/nic/update?hostname=myhostname`, this is going through your container default DNS. If for whatever reason your Pihole configuration doesn't work, just set `--dns 1.1.1.1` or `dns: 1.1.1.1` in your container run command/docker-compose.yml to use 1.1.1.1 for the entire container.

> I do see the requests to "dyndns.strato.com" and they are not blocked, so general connection between my DDNS-updater and my pihole seems to work as well.

Now, as to why pihole produces a connection refused error, I have no idea unfortunately.

Thanks!

**Reply by timkolloch** [reply]
> `RESOLVER_ADDRESS` is actually only used to resolve your domain names, not for anything else. Readme improved for this in 99d670f7f992e3db69edecd403b3f209053fb5dd.
> 
> 
> 
> So in your case `mydomain:***@dyndns.strato.com/nic/update?hostname=myhostname`, this is going through your container default DNS. If for whatever reason your Pihole configuration doesn't work, just set `--dns 1.1.1.1` or `dns: 1.1.1.1` in your container run command/docker-compose.yml to use 1.1.1.1 for the entire container.
> 
> 
> 
> > I do see the requests to "dyndns.strato.com" and they are not blocked, so general connection between my DDNS-updater and my pihole seems to work as well.
> 
> 
> 
> Now, as to why pihole produces a connection refused error, I have no idea unfortunately.
> 
> 
> 
> Thanks!

Oh okay I get that now, thank you. Setting the DNS for the whole container works for now haha. Guess it will stay like that, thanks for the clarification, have a nice evening!
